src/forms/FormFieldUpload/FormFieldUpload.tsx

```diff
@@ -200,8 +200,14 @@ const FormFieldUpload = (props: MosaicFieldProps<"upload", UploadFieldInputSetti
 	};
 
 	const shouldDisableField = useMemo(() => {
-		return fieldDef.disabled || (limit !== undefined && value?.length >= limit)
-	}, [fieldDef.disabled, limit, value]);
+		const numOfPendingFiles = Object.keys(pendingFiles).length;
```

Comment:
The pending files object also includes those that have error'd, this means the field should get disabled **only** when the files are loading and only take into account those that don't have error messages. For example, let's assume the file limit is 3, the user uploads 3 files, while they're loading the filed is disabled, then all finish loading but 1 throws an error message, with the code above the field would still be disabled.
 ```suggestion
		const numOfPendingFiles = Object.values(pendingFiles).filter((pendingFile: {error: string}) => pendingFile.error === undefined).length;
```